sqlglot/dialects/dialect.py

```diff
@@ -256,36 +283,71 @@ def format_time(
 
         return expression
 
-    @classmethod
-    def normalize_identifier(cls, expression: E) -> E:
+    def __init__(self, **kwargs) -> None:
+        normalization_strategy = kwargs.get("normalization_strategy")
+
+        if normalization_strategy is None:
+            self.normalization_strategy = self.NORMALIZATION_STRATEGY
+        else:
+            self.normalization_strategy = NormalizationStrategy(normalization_strategy.upper())
+
+    def __eq__(self, other: t.Any) -> bool:
+        # Does not currently take dialect state into account
+        return type(self) == other
+
+    def __hash__(self) -> int:
+        # Does not currently take dialect state into account
+        return hash(type(self))
```

Comment:
@tobymao can you take a look at these two methods? What do you think their behavior should be?

I tried to change `__eq__` to take `self.normalization_strategy` into account but then if we wanted to change `__hash__` to match that, `Snowflake() in {"snowflake"}` would yield `False` (i.e. we have tests that failed)

Reply:
Chatted in Slack: this behavior is fine for now, i.e. we won't take dialect state into account when comparing.